src/pytest_bdd/scenario.py

```diff
@@ -303,12 +305,26 @@ def scenario_wrapper(request: FixtureRequest, _pytest_bdd_example: dict[str, str
 def collect_example_parametrizations(
     templated_scenario: ScenarioTemplate,
 ) -> list[ParameterSet] | None:
-    if templated_scenario.examples is None:
-        return None
-    if contexts := list(templated_scenario.examples.as_contexts()):
-        return [pytest.param(context, id="-".join(context.values())) for context in contexts]
-    else:
-        return None
+    parametrizations = []
+
+    for examples in templated_scenario.examples:
+        tags: set = examples.tags or set()
+
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=PytestUnknownMarkWarning)
```

Comment:
why do we need to silence warnings?

Reply:
My mistake - for some reason I thought the existing hook was suppressing warnings for marks registered from Gherkin tags.  I have now removed this code :)